src/lib/object-diff.js

```diff
@@ -3,25 +3,26 @@ var assert = require('assert')
 function difference(a, b) {
   var diff = {}
 
-  Object.keys(a).forEach(compare(diff, a, b))
-  Object.keys(b).forEach(compare(diff, a, b))
+  var aKeys = Object.keys(a)
+  var bKeys = Object.keys(b)
 
-  return diff
-}
-
-function compare(diff, a, b) {
-  return function curried(n) {
-    if (!diff[n]) {
-      try {
-        assert.deepEqual(a[n], b[n])
-      } catch (e) {
-        diff[n] = {
-          config1: a[n],
-          config2: b[n],
-        }
+  function compare(n, index) {
+    try {
+      assert.deepEqual(a[n], b[n])
+    } catch (e) {
+      diff[n] = {
+        config1: a[n],
+        config2: b[n],
       }
+      aKeys.splice(index, 0)
+      bKeys.splice(index, 0)
```

Comment:
That wont work. [`deleteCount` is zero, so the splice is actually doing nothing](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/splice).
If you drop elements this way during iteration (regardless of whether using `forEach` or a classic `for` loop), you mess up the whole loop.

Sample Code:

```JavaScript
var arr = [1,2,3,4,5,6,7,8,9,0];
arr.forEach((n, i) => {
  console.log(n, i);
  arr.splice(i, 1);
});
```

Output:

``
1 0
3 1
5 2
7 3
9 4
```